src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -333,17 +336,27 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             this.regionPlace = RegionPlace.Before;
             EnterParameters(methodParameters);               // with parameters assigned
 
-            if (_symbol is (not SynthesizedPrimaryConstructor) and { ContainingSymbol: SourceMemberContainerTypeSymbol { PrimaryConstructor: { } primaryConstructor } })
+            switch (_symbol)
             {
-                // All primary constructor parameters are definitely assigned outside of the primary constructor
-                foreach (var parameter in primaryConstructor.Parameters)
-                {
-                    int slot = GetOrCreateSlot(parameter);
-                    if (slot > 0)
+                case MethodSymbol { IsStatic: false, ContainingSymbol: SourceMemberContainerTypeSymbol { PrimaryConstructor: { } primaryConstructor } } and
+                     (not SynthesizedPrimaryConstructor):
                     {
-                        SetSlotState(slot, true);
+                        var save = CurrentSymbol;
+                        CurrentSymbol = primaryConstructor;
+
+                        // All primary constructor parameters are definitely assigned outside of the primary constructor
+                        foreach (var parameter in primaryConstructor.Parameters)
+                        {
+                            NoteWrite(parameter, value: null, read: true);
+                        }
+
+                        CurrentSymbol = save;
                     }
-                }
+                    break;
+
+                case (FieldSymbol or PropertySymbol) and { IsStatic: false, ContainingSymbol: SourceMemberContainerTypeSymbol { PrimaryConstructor: { } primaryConstructor } }:
```

Comment:
>case [](http://example.com/codeflow?start=16&length=4)

Is this case doing the same work as the case above? Could the two cases be combined?

Reply:
> Is this case doing the same work as the case above? 

No, they are doing different things. Initializers are considered to be inside the primary constructor. Methods are not.